base_python/python3.8/Dockerfile

```diff
@@ -0,0 +1,10 @@
+FROM python:3.8-slim as compile-image
+
+ARG DEBIAN_FRONTEND=noninteractive
+
+RUN apt-get update
+RUN apt-get install -y --no-install-recommends \
```

Comment:
I would add both of these in one run command, in case a package from the second RUN changes, docker would rebuild the image from second RUN statement but it will operate on the same metadata it fetched from the older somewhat stale apt-get update command as it would be cached.

Good example to see this would be some production dockerfiles like the from ghost library here https://github.com/docker-library/ghost/blob/03e10bf5352f44ed78525d6a7b4d4096d55b63fc/5/debian/Dockerfile#L8